notifier.go

```diff
@@ -21,11 +21,19 @@ type Notifier struct {
 	cfg *Configuration
 
 	// sessions
-	sessionChannel         chan *session
+	sessionCh              chan *session
 	sessions               []*session
-	sessionMutex           sync.Mutex
 	sessionPublishInterval time.Duration
-	sessionOnce            sync.Once
+
+	loopOnce sync.Once
+
+	reportCh   chan *report
```

Comment:
Ordering within this struct could be better (match the constructor at the very least)